lightning/src/ln/channelmanager.rs

```diff
@@ -7114,35 +7114,25 @@ where
 		log_debug!(self.logger, "Generating channel_reestablish events for {}", log_pubkey!(counterparty_node_id));
 
 		let per_peer_state = self.per_peer_state.read().unwrap();
-		for (_cp_id, peer_state_mutex) in per_peer_state.iter() {
+		if let Some(peer_state_mutex) = per_peer_state.get(counterparty_node_id) {
 			let mut peer_state_lock = peer_state_mutex.lock().unwrap();
 			let peer_state = &mut *peer_state_lock;
 			let pending_msg_events = &mut peer_state.pending_msg_events;
 			peer_state.outbound_v1_channel_by_id.retain(|_, chan| {
-				if chan.context.get_counterparty_node_id() == *counterparty_node_id {
-					// If we created this (outbound) channel while we were disconnected from the
-					// peer we probably failed to send the open_channel message, which is now
-					// lost. We can't have had anything pending related to this channel, so we just
-					// drop it.
-					chan.context.have_received_message()
-				} else { true }
+				// We shouldn't have any pending outbound channels upon reconnect. If one exists,
+				// then we probably tried to open a channel while the peer was disconnected and they
+				// wouldn't have received the corresponding channel open message. We can just force-close
+				// these channels and drop them.
+				log_error!(self.logger, "Force-closing channel {}", log_bytes!(chan.context.channel_id()));
+				self.issue_channel_close_events(&chan.context, ClosureReason::HolderForceClosed);
+				self.finish_force_close_channel(chan.context.force_shutdown(false));
+				false
```

Comment:
Should we just `clear` after `iter`ating instead of using `retain`?

Reply:
Yeah that’s more idiomatic since we never retain.